PowerEditor/src/functionList.xml

```diff
@@ -123,21 +123,21 @@
 						<nameExpr expr="\w+\Z" />
 					</className>
 					<function
-						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch))[\w~]+\s*\([^\{]*\{"
+						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch|foreach))[\w~]+\s*\([^\{]*\{"
 					>
 						<functionName>
-							<funcNameExpr expr="(?!(if|while|for|switch))\w+\s*\([^\{]*" />
+							<funcNameExpr expr="(?!(if|while|for|switch|foreach))\w+\s*\([^\{]*" />

```

Comment:
Should we also add [`do`](https://www.php.net/manual/en/control-structures.do.while.php)?

Reply:
However, I can't reproduce a false positive happening with these structures.

Could you please provide an example showing the need for these regex negative assertions?

Reply:
@vlakoff  function list in `futures` - not all work in curret version. See https://notepad-plus-plus.org/features/function-list.html